src/Castle.Windsor/MicroKernel/Resolvers/DefaultDependencyResolver.cs

```diff
@@ -361,12 +361,7 @@ private object ResolveFromKernelByName(CreationContext context, ComponentModel m
 
 		private object ResolveFromKernelByType(CreationContext context, ComponentModel model, DependencyModel dependency)
 		{
-			IHandler handler;
-			try
-			{
-				handler = TryGetHandlerFromKernel(dependency, context);
-			}
-			catch (HandlerException exception)
+			if (!TryGetHandlerFromKernel(dependency, context, out var handler))
```

Comment:
The `return false` case looks almost identical to the `handler = null; return true` case below it.
There's a potentially a better factoring to indicate which of the two messages should be used?

Reply:
⚠ Oh, this may not be a valid refactoring. I forgot to consider HandlerExceptions that may be thrown by methods called by `TryGetHandlerFromKernel`. We might need to catch `HandlerException` within `TryGetHandlerFromKernel` and return false;